waku/v2/node/storage/message/waku_message_store.nim

```diff
@@ -12,22 +13,49 @@ import
 
 export sqlite
 
+logScope:
+  topics = "wakuMessageStore"
+
 const TABLE_TITLE = "Message"
+const maxStoreOverflow = 1.3
 
 # The code in this file is an adaptation of the Sqlite KV Store found in nim-eth.
 # https://github.com/status-im/nim-eth/blob/master/eth/db/kvstore_sqlite3.nim
-#
-# Most of it is a direct copy, the only unique functions being `get` and `put`.
 
 type
+  # WakuMessageStore implements auto deletion as follows:
+  # The sqlite DB will store up to `storeCapacity` * `maxStoreOverflow` messages.
+  # If more messages are stored, the messages are sorted by `receiverTimestamp` and the oldest ones are deleted,
+  # bringing the total number of stored messages back to `storeCapacity*maxStoreOverflow/2`. The rationale for batch deleting is efficiency.
```

Comment:
> bringing the total number of stored messages back to `storeCapacity*maxStoreOverflow/2`. The rationale for batch deleting is efficiency. 

I assume the `storeCapacity` indicates the in-memory limit, right? if yes, then I think the `maxStoreOverflow` should be a value larger than `2`, otherwise, while truncating, you may end up deleting DB messages that are still available in the memory. 

For example, for `maxStoreOverflow = 1.3`, the persisted db messages will be `storeCapacity`*0.65, which means the db does not contain all the in-mem messages.

Reply:
Right, if the idea is to only remove half of the _overflow_ (not half of the entire capacity), shouldn't the formula for total messages after deletion be something like `storeCapacity*(1 + (maxStoreOverflow - 1)/2)`. A bit of a complicated way to express this though. Perhaps it's easier to just determine the absolute number of overflow messages allowed `(maxStoreOverflow*storeCapacity - storeCapacity)` when initiating the store and always removing half of that?

Reply:
@staheri14 Yes. Good point. The store does not sync between DB and in-memory store.
This is also one of the points we plan to address in the future.
The currently implemented formula is as @jm-clius stated.
Only half of the overflow window gets deleted. The full `storeCapacity` and half of the overflow window are kept.

Reply:
> Perhaps it's easier to just determine the absolute number of overflow messages allowed (maxStoreOverflow*storeCapacity - storeCapacity) when initiating the store and always removing half of that?

I now added a `deleteWindow` to `WakuMessageStore`.